### PR TITLE
8292448: Convert BitMapFragmentTable to ResourceHashtable

### PR DESCRIPTION
**Pull request by coleenp**
This is an uncomplicated conversion of the bitmap fragment table into a ResizeableResourceHashtable.  Ran tiers 1-3 and 4-7 with other changes, and verified that the memory for ResourceHashtableStorage is freed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292448](https://bugs.openjdk.org/browse/JDK-8292448): Convert BitMapFragmentTable to ResourceHashtable

### Reviewers
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**)
 * [Robbin Ehn](https://openjdk.org/census#rehn) (@robehn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9893/head:pull/9893` \
`$ git checkout pull/9893`

Update a local copy of the PR: \
`$ git checkout pull/9893` \
`$ git pull https://git.openjdk.org/jdk pull/9893/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9893`

View PR using the GUI difftool: \
`$ git pr show -t 9893`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9893.diff">https://git.openjdk.org/jdk/pull/9893.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/utilities/objectBitSet.inline.hpp [hunk 1]
```diff
@@ -53,52 +52,15 @@ ObjectBitSet<F>::~ObjectBitSet() {
...
+    public:
+    bool do_entry(uintptr_t& key, CHeapBitMap*& fragment) {
+      // fragment is already deleted above, but now delete the table nodes.
+      return true;
     }
-  }
-  return NULL;
+  };
+  Deleter deleter;
+  _bitmap_fragments.unlink(&deleter);
```
Comment (iklam) [inline src/hotspot/share/utilities/objectBitSet.inline.hpp:63]: This shouldn't be necessary. The destructor of _bitmap_fragments will be implicitly called and will free all of its entries.

**Review by iklam — COMMENTED** [review]
Looks good, but I think the explicit delete is not necessary.

File: src/hotspot/share/utilities/objectBitSet.inline.hpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/share/utilities/objectBitSet.inline.hpp:63]: Yes, I added a temporary assert and indeed it is.  Thanks for pointing this out.

Comment (coleenp) [reply src/hotspot/share/utilities/objectBitSet.inline.hpp]: I'd actually tested with the version without it and then thought I needed it this morning, so I'm glad I don't need it.

**Review by iklam — APPROVED** [review]
Looks good. A lot of lines removed!

**Review by robehn — APPROVED** [review]
Thanks!

**Comment by coleenp** [comment]
Thanks for the reviews, Ioi and Robbin!
/integrate

**Pull request closed without merging**